src/main/java/org/folio/rest/impl/TenantPermsAPI.java

```diff
@@ -62,19 +77,396 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
     }
   }
 
-  private Future<Void> savePermList(List<Perm> permList, Context vertxContext, String tenantId) {
+  private Future<List<Permission>> getPermsForModule(ModuleId moduleId, List<OkapiPermission> permSet,
+      Context vertxContext, String tenantId) {
+    Promise<List<Permission>> promise = Promise.promise();
+    
+    try {
+      getPermsByModule(moduleId, vertxContext, tenantId)
+        .onComplete(getResult -> {
+          if (getResult.failed()) {
+            Throwable cause = getResult.cause();
+            logger.error(cause.getMessage(), cause);
+            promise.fail(cause);
+            return;
+          }
+
+          List<Permission> existing = getResult.result();
+          if (existing.isEmpty() && permSet != null) {
+            // this means either:
+            // A) the first time enabling this module, or
+            // B) the permissions exist but don't yet have the definedBy field
+            List<OkapiPermission> perms = permSet.stream()
+                .filter(p -> p.getPermissionName() != null)
+                .collect(Collectors.toList());
+            List<Permission> ret = new ArrayList<>();
+
+            List<Future> futures = new ArrayList<>(perms.size());
```

Comment:
Use proper generics with https://github.com/folio-org/okapi/blob/master/okapi-common/src/main/java/org/folio/okapi/common/GenericCompositeFuture.java

Reply:
We can't do this without pulling in newer versions of okapi-common and vertx.  Our best bet is to "fix" this in #108

Reply:
@MikhailFokanov FYI